packages/base/src/features/patchPopup.ts

```diff
@@ -94,9 +124,7 @@ const patchClosed = (Popup: OpenUI5Popup) => {
 const patchFocusEvent = (Popup: OpenUI5Popup) => {
 	const origFocusEvent = Popup.prototype.onFocusEvent;
 	Popup.prototype.onFocusEvent = function onFocusEvent(e: FocusEvent) {
-		// If the popup is the topmost one, we call the original focus event handler from the OpenUI5 Popup,
-		// otherwise the focus event is handled by the Web Component Popup.
-		if (this === getTopmostPopup()) {
+		if (shouldCallOpenUI5FocusEvent(this)) {
```

Comment:
an alternative is to patch only the logic for topmost popup - see https://github.com/SAP/ui5-webcomponents/commit/54781ca5d3e27d0f6c381a45abac6b4ac999dabe and https://git.wdf.sap.corp/c/openui5/+/6794573
unfortunately most issues still remain, but seems better patching for long term solution as it directly fixes the initial problem